docs/installation.md

```diff
@@ -2,50 +2,73 @@
+- **dune**: [`dune`](https://github.com/ocaml/dune) is the official build system in the OCaml ecosystem. Check out the [manual](https://dune.readthedocs.io/en/latest/) for more details on how to set up your project.
 
-## New Project
+All your packages are managed in your `package.json` file. Usually you will find a `dune` file in each source code directory (such as `bin/` and `lib/`) for all the build system settings as well.
 
-The global installation comes with a simple project generator. Try:
 
-```sh
-bsb -init my-new-project -theme basic-reason
+### Some `esy` Tips
+
+**Adding dependencies:**
+
+To add Reason / OCaml packages that happen to be hosted on npm, run `esy add npm-package-name`.
+
+```
+esy add refmterr
 ```
 
-To compile & run the project you just created:
+**Opam integration:**
 
-```sh
-cd my-new-project
-yarn build # or npm run build, for npm
-node src/Demo.bs.js
+`esy` treats the npm scope `@opam` specially. `esy` will install any package name with the `@opam` scope directly from [opam](https://opam.ocaml.org/packages/). This is the only scope with special meaning. All other package names are assumed to be hosted on npm.
+
+```
+esy add @opam/bos
 ```
 
-That uses BuckleScript to compile Reason to JavaScript, then uses NodeJS to run the JavaScript. Feel free to use the generated JS files in whichever way you'd like, as if they're hand-written by you.
+**Advanced esy configuration:**
 
-During development, instead of running `npm run build` each time to compile, run `npm run start` to start a watcher that recompiles automatically after file changes.
+See the [configuration](https://esy.sh/docs/en/configuration.html) section from the complete `esy` docs.
 
-By default, the `basic-reason` theme configures BuckleScript to output the generated JS files alongside the Reason files they were produced from. [There's a reason why this is helpful](https://bucklescript.github.io/docs/en/build-overview#tips-tricks). If you prefer to keep the generated files somewhere else, edit `bsconfig.json` to set `in-source` to `false`; the JS files will then be output to the `lib/js` directory instead.
+## Compiling to JavaScript
 
-Alternatively, **to start a [ReasonReact](https://reasonml.github.io/reason-react/docs/en/installation.html) app**, follow the instructions [here](https://reasonml.github.io/reason-react/docs/en/installation).
+Reason + OCaml both leverage the [js_of_ocaml (JSOO)](https://ocsigen.org/js_of_ocaml/3.7.0/manual/overview) compiler to compile from bytecode to JavaScript.
 
-## Existing Project
+To get started with Reason + esy + JSOO, check out this [`hello-jsoo-esy`](https://github.com/jchavarri/hello-jsoo-esy) template:
 
-You can install the toolchain locally to an existing project, through the familiar command:
+```
+git clone https://github.com/jchavarri/hello-jsoo-esy.git
+cd hello-jsoo-esy
+esy
 
-```sh
-yarn add --dev bs-platform
+yarn
```

Comment:
change to `npm`? Kind of weird that above we use `npm` to globally install `esy`, but here we've switched to `yarn`.

Reply:
done